From 01ac40247d502e15ee657277dd5a5d31486f7831 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 29 Aug 2021 02:31:44 +0200
Subject: [PATCH 286/467] mfd: rk808: Add support for power off on RK817

RK817 has a power-off bit in SYS_CFG3. Add support for powering
off the PMIC.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/mfd/rk808.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index 77ccd31ca..b181fe401 100644
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -543,6 +543,10 @@ static void rk808_pm_power_off(void)
 		reg = RK808_DEVCTRL_REG,
 		bit = DEV_OFF_RST;
 		break;
+	case RK817_ID:
+		reg = RK817_SYS_CFG(3);
+		bit = DEV_OFF;
+		break;
 	case RK818_ID:
 		reg = RK818_DEVCTRL_REG;
 		bit = DEV_OFF;
-- 
2.34.0

